### ISSUE TITLE
compliance-checker ugrid plugin

### ISSUE THREAD

**Issue by ocefpaf** [ioos/ioos-code-sprint#6]
Project Description:

IOOS developed a [UGRID checks plugin](https://github.com/ioos/cc-plugin-ugrid) for compliance-checker and recently the UK Met Office developed a similar tool, [ugrid-checks](https://github.com/pp-mo/ugrid-checks).

The project would be more of a discussion on how to integrate the two efforts. One solution would be to halt the ugrid plugin development in lieu for the newer ugrid-checks. The Software challenge here the different nature of both Software where one is a plugin and the other stand-alone. Deprecating the plugin would cause some friction to its current users.

Expected Outcomes: 

A unified ugrid checker that, hopefully, works with compliance-checker.

Skills required:

Familiarity with either package code base and knowledge of the [ugrid standards](https://ugrid-conventions.github.io/ugrid-conventions/).

Difficulty:

Low. While hacking on the code and fixing the problem would be nice the main goal is to write a roadmap for the ugrid checker.

Topic Lead(s):

Ping here @bjlittle and @pp-mo from the UK Met Office hoping they can participate and @benjwadams who is the lead on the compliance-checker track.

Relevant links:

All inline above.

**Comment by pp-mo** [comment]
Hi @ocefpaf 

I'd be interested to know how we/I might  contribute to next week's sprint?

Since it's getting rather late now (!), I ought to feedback on what I've been considering here regarding the relationship between [cc-plugin-ugrid](https://github.com/ioos/cc-plugin-ugrid)  and [ugrid-checks](https://github.com/pp-mo/ugrid-checks) ...

I have tried to gain an overview understanding of the IOOS checker framework, and I've taken a close look at the existing cc-ugrid-plugin general, and compared the scope of the actual checks it implements with those here.

For ref, my slightly random observations-so-far, [noted here](https://github.com/ioos/cc-plugin-ugrid/pull/17).

I have also been considering what it would take to make the ugrid-checks code work within the context of an ioos plugin.
Still work-in-progress on that, but some quick ideas :
  1. ioos applies a 3-step "severity level" for tests, whereas ugrid-checks uses only 2 (~ error/warning).
    No big problem with that
  1. ioos expects a test to identify checks which were run ok, _as well as_ checks that failed
  1. **_ideally_**, an ioos test will return a composite result listing multiple of checks-run and passed/failed
  1. where a test is re-run in a different context (e.g. applied to a different variable of the same role), ioos expects to generate a result that summarises N-passed / N-checked.

In short, I've been drafting some code changes that might help us bridge this function gap, and potentially may even improve the code structure in the ugrid-checks repo.

.. **FYI**, in haste, more to follow (will re-edit this later) ..

**Comment by pp-mo** [comment]
### Quick update : 
I have posted a quick note on [a problem I uncovered in the main cf-checker](https://github.com/ioos/compliance-checker/issues/922).
I think I could provide some similar observations on fixes to the [cc-ugrid-plugin](https://github.com/ioos/cc-plugin-ugrid), if useful ?  (various problems spotted, as noted above)

In the meantime, as I wrote on the [ugrid-checks isuue matching this](https://github.com/ioos/cc-plugin-ugrid/pull/17) ...
> ### Regarding possible use of the ugrid-checks code
>
> I have already created a proof-of-concept solution (incomplete, and not yet visible here) to provide checking results in a suitable form for inclusion in an IOOS-style Report object.
So, if anyone is interested, I think we could collaborate on creating an ioos plugin to integrate ugrid-checks via such an interface.
It makes sense to me, that we would want the plugin in its own project/repo. But of course, the sprint may simply not be a very good time, since the time difference can be a bit of an obstacle.
Also, I still remain to be convinced that this is particularly wanted -- but the interfacing problem did pique my interest !
